### PR TITLE
[Fix-14721] [K8S Task] Handle job delete event when user manually delete the job in k8s cluster

### LINKED ISSUE #14721: [Bug] [K8S Task] Handle job delete event when user manually delete the job in k8s cluster
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

Run a K8S task, if the pod is pending for some reasons (such as unmatchable node label), and then user manually delete k8s job in the cluster via command line -  `kubectl delete job`. Now, DS can not handle job delete event, and job will keep running.

### What you expected to happen

We should better handle this delete action and set the task status to failure, directly delete job may because of some unexpected operation.

### How to reproduce

Ditto.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Radeity**
## Purpose of the pull request
- close https://github.com/apache/dolphinscheduler/issues/14721

## Brief change log
- Handle `Delete` action, and fail the k8s task.

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/impl/K8sTaskExecutor.java [hunk 1]
```diff
@@ -290,7 +294,7 @@ public TaskResponse run(String k8sParameterStr) throws Exception {
                     // Wait kubernetes pod log collection finished
                     podLogOutputFuture.get();
                 } catch (ExecutionException e) {
-                    log.error("Handle pod log error", e);
```
Comment (qingwli) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/impl/K8sTaskExecutor.java]: Whether if we print exceptions in the log better?

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by qingwli — APPROVED** [review]
+1

**Review by Gallardot — APPROVED** [review]
LGTM

**Pull request merged**